src/test/java/edu/ucla/library/iiif/fester/handlers/PutCollectionHandlerTest.java

```diff
@@ -0,0 +1,212 @@
+import java.util.UUID;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.amazonaws.SdkClientException;
+
+import edu.ucla.library.iiif.fester.Config;
+import edu.ucla.library.iiif.fester.Constants;
+import edu.ucla.library.iiif.fester.HTTP;
+import edu.ucla.library.iiif.fester.MessageCodes;
+import edu.ucla.library.iiif.fester.utils.IDUtils;
+import info.freelibrary.util.Logger;
+import info.freelibrary.util.LoggerFactory;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.RequestOptions;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+
+public class PutCollectionHandlerTest extends AbstractFesterHandlerTest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PutCollectionHandlerTest.class, Constants.MESSAGES);
+
+    private String myPutCollectionID;
+
+    private String myPutCollectionS3Key;
+
+    /**
+     * Test set up.
+     *
+     * @param aContext A testing context
+     */
+    @Override
+    @Before
+    public void setUp(final TestContext aContext) throws IOException {
+        super.setUp(aContext);
+
+        final String putPrefix = "PUT_";
+
+        myPutCollectionID = putPrefix + UUID.randomUUID().toString();
+        myPutCollectionS3Key = IDUtils.getCollectionS3Key(myPutCollectionID);
+    }
+
+    /**
+     * Test tear down.
+     *
+     * @param aContext A testing context
+     */
+    @Override
+    @After
+    public void tearDown(final TestContext aContext) {
+        super.tearDown(aContext);
+
+        try {
+            // If object doesn't exist, this still completes successfully
+            myS3Client.deleteObject(myS3Bucket, myPutCollectionS3Key);
+        } catch (final SdkClientException details) {
+            aContext.fail(details);
+        }
+
+        myVertx.close(aContext.asyncAssertSuccess());
+    }
+
+    /**
+     * Test the PutCollectionHandler.
+     *
+     * @param aContext A testing context
+     */
+    @Test
+    @SuppressWarnings("deprecation")
+    public void testPutCollectionHandler(final TestContext aContext) throws IOException {
+        final String manifestPath = COLLECTION_FILE.getAbsolutePath();
+        final Buffer manifest = myVertx.fileSystem().readFileBlocking(manifestPath);
+        final Async asyncTask = aContext.async();
+        final int port = aContext.get(Config.HTTP_PORT);
+        final String requestPath = IDUtils.getResourceURIPath(myPutCollectionS3Key);
+        final RequestOptions requestOpts = new RequestOptions();
+
+        LOGGER.debug(MessageCodes.MFS_016, requestPath);
+
+        requestOpts.setPort(port).setHost(Constants.UNSPECIFIED_HOST).setURI(requestPath);
+        requestOpts.addHeader(Constants.CONTENT_TYPE, Constants.JSON_MEDIA_TYPE);
+
+        myVertx.createHttpClient().put(requestOpts, putResponse -> {
+            final int putStatusCode = putResponse.statusCode();
+
+            switch (putStatusCode) {
```

Comment:
Not a big deal, just wondering why a `switch` here instead of an `if` / `else` with just the two choices?

Reply:
I think when I started editing this code, I had 3 case blocks, but now that there are only two it makes sense to use a simple if/else.